From b51459d548836cb14b11a3cd5bb631f5c1fb7c77 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sat, 5 Jan 2019 21:03:18 +0100
Subject: [PATCH 1/1] sandbox: i2c_emul_find() No emulators for device 'rtc@43'

when running the date command on sandbox_defconfig an error occurs:

    ./u-boot -D u-boot.dtb

    => date
    i2c_emul_find() No emulators for device 'rtc@43'
    ## Get date failed

Correct the references to the emulator devices in the sandbox device trees
using test.dts as a reference.

Fixes: 031a650e1309 ("dm: sandbox: i2c: Use new emulator parent uclass")
Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 arch/sandbox/dts/sandbox.dts   | 8 ++++----
 arch/sandbox/dts/sandbox64.dts | 6 ++++--
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/arch/sandbox/dts/sandbox.dts b/arch/sandbox/dts/sandbox.dts
index ae3189ec8c..5837dd2d26 100644
--- a/arch/sandbox/dts/sandbox.dts
+++ b/arch/sandbox/dts/sandbox.dts
@@ -100,11 +100,13 @@
 		eeprom@2c {
 			reg = <0x2c>;
 			compatible = "i2c-eeprom";
+			sandbox,emul = <&emul_eeprom>;
 		};
 
 		rtc_0: rtc@43 {
 			reg = <0x43>;
 			compatible = "sandbox-rtc";
+			sandbox,emul = <&emul0>;
 		};
 		sandbox_pmic: sandbox_pmic {
 			reg = <0x40>;
@@ -119,14 +121,12 @@
 			#size-cells = <0>;
 			reg = <0xff>;
 			compatible = "sandbox,i2c-emul-parent";
-			emul-eeprom {
-				reg = <0x2c>;
+			emul_eeprom: emul-eeprom {
 				compatible = "sandbox,i2c-eeprom";
 				sandbox,filename = "i2c.bin";
 				sandbox,size = <256>;
 			};
-			emul0 {
-				reg = <0x43>;
+			emul0: emul0 {
 				compatible = "sandbox,i2c-rtc";
 			};
 		};
diff --git a/arch/sandbox/dts/sandbox64.dts b/arch/sandbox/dts/sandbox64.dts
index d30fd62a2a..a3c95f2cdb 100644
--- a/arch/sandbox/dts/sandbox64.dts
+++ b/arch/sandbox/dts/sandbox64.dts
@@ -90,11 +90,13 @@
 		eeprom@2c {
 			reg = <0x2c>;
 			compatible = "i2c-eeprom";
+			sandbox,emul = <&emul_eeprom>;
 		};
 
 		rtc_0: rtc@43 {
 			reg = <0x43>;
 			compatible = "sandbox-rtc";
+			sandbox,emul = <&emul0>;
 		};
 		sandbox_pmic: sandbox_pmic {
 			reg = <0x40>;
@@ -107,12 +109,12 @@
 		i2c_emul: emul {
 			reg = <0xff>;
 			compatible = "sandbox,i2c-emul-parent";
-			emul-eeprom {
+			emul_eeprom: emul-eeprom {
 				compatible = "sandbox,i2c-eeprom";
 				sandbox,filename = "i2c.bin";
 				sandbox,size = <256>;
 			};
-			emul0 {
+			emul0: emul0 {
 				compatible = "sandbox,i2c-rtc";
 			};
 		};
-- 
2.20.1

